src/components/LanguageSelection/LanguageSelection.js

```diff
@@ -0,0 +1,104 @@
+import React, { useContext, useState } from "react";
+import { ConfigContext } from "Context";
+import { useAsync } from "react-use";
+import Loader from "components/Loader";
+import { FormattedMessage, useIntl } from "react-intl";
+import { v4 as uuidv4 } from "uuid";
+import {
+  ButtonGroup,
+  Heading,
+  LinkButton,
+} from "@utrecht/component-library-react";
+import { get, put } from "../../api";
+import PropTypes from "prop-types";
+
+const propTypes = PropTypes.shape({
+  heading: PropTypes.string,
+  headingLevel: PropTypes.number,
+  onLanguageChange: PropTypes.func,
+});
+
+const LanguageSelection = (props) => {
+  // Hook uses
+  const config = useContext(ConfigContext);
+  const { locale } = useIntl();
+  const [current, setCurrent] = useState(locale);
+  const getAvailableLanguages = async () => {
+    const result = await get(`${config.baseUrl}i18n/info`);
+    setCurrent(result.current); // state change if current changed
+    return result.languages.map(({ code, name }) => ({
+      lang: code,
+      textContent: code.toUpperCase(),
+      label: name,
+      current: code === result.current,
+    }));
```

Comment:
This transform should be moved outside of the hook - it's side-effect free and  it allows you to use other hook return values/state (for the `current` bool) without having to pass that to the hook dependencies. Eventually we'll then be able to avoid hitting the API when it's not absolutely needed.